nesta_ds_utils/file_ops.py

```diff
@@ -1,53 +1,196 @@
-    :type out_path: str or pathlib.Path
-    :param delete_zip: option to delete zip file after extracted
-    :type delete_zip: Boolean
-    """
 
+    Args:
+        zip_path (Union[Path, str]): Path to zipped file.
+        out_path (Union[Path, str], optional): Path where contents will be unzipped to. Defaults to None.
+        delete_zip (Boolean, optional): Option to delete zip file after extracted. Defaults to True.
+    """
     if out_path is None:
         out_path = zip_path.rsplit("/", 1)[0]
 
     make_path_if_not_exist(out_path)
     with zipfile.ZipFile(zip_path, "r") as z:
         z.extractall(out_path)
-    
+
     if delete_zip is True:
-        os.remove(zip_path)
\ No newline at end of file
+        os.remove(zip_path)
+
+
+def get_bucket_filenames_s3(bucket_name: str, dir_name: str = "") -> List[str]:
+    """Get a list of all files in bucket directory.
+
+    Args:
+        bucket_name (str): S3 bucket name
+        dir_name (str, optional): Directory or sub-directory within an S3 bucket.
+            Defaults to '' (the top level bucket directory).
+
+    Returns:
+        List[str]: List of file names in bucket directory.
+    """
+    s3_resources = boto3.resource("s3")
+    my_bucket = s3_resources.Bucket(bucket_name)
+    return [
+        object_summary.key
+        for object_summary in my_bucket.objects.filter(Prefix=dir_name)
+    ]
+
+
+def _df_to_fileobj(df: pd.DataFrame, save_file_dir: str) -> io.BytesIO:
+    """Convert DataFrame into bytes file object.
+
+    Args:
+        df (pd.DataFrame): Dataframe to convert.
+        save_file_dir (str): Saving file name.
+
+    Returns:
+        io.BytesIO: Bytes file object.
+    """
+    buffer = io.BytesIO()
+    if fnmatch(save_file_dir, "*.csv"):
+        df.to_csv(buffer)
+    elif fnmatch(save_file_dir, "*.json"):
+        df.to_json(buffer)
+    elif fnmatch(save_file_dir, "*.pkl"):
+        df.to_pickle(buffer)
+    elif fnmatch(save_file_dir, "*.xml"):
+        df.to_xml(buffer)
+    else:
+        raise Exception(
+            "Automatic convertion from pd.DataFrame currently supported only for 'csv', 'json', 'pkl' and 'xml'."
+        )
+    buffer.seek(0)
+    return buffer
+
+
+def _fileobj_to_df(fileobj: io.BytesIO, load_file_dir: str) -> pd.DataFrame:
+    """Convert bytes file object into DataFrame.
+
+    Args:
+        buffer (io.BytesIO): Bytes file object.
+        load_file_dir (str): Loading file name.
+
+    Returns:
+        pd.DataFrame: Dataframe converted
+    """
+    if fnmatch(load_file_dir, "*.csv"):
+        df = pd.read_csv(fileobj)
+    elif fnmatch(load_file_dir, "*.json"):
+        df = pd.read_json(fileobj)
+    elif fnmatch(load_file_dir, "*.pkl"):
+        df = pd.read_pickle(fileobj)
+    elif fnmatch(load_file_dir, "*.xml"):
+        df = pd.read_xml(fileobj)
+    else:
+        raise Exception(
+            "Automatic convertion into pd.DataFrame currently supported only for 'csv', 'json', 'pkl' and 'xml'."
```

Comment:
After the AI and genomics project, we decided to shift to using parquet files as they have such good compression. Can we add handling for that? This will mean adding `pyarrow` as a dependency.

Reply:
Sure